src/main/java/seedu/recipe/ui/PlanningListCard.java

```diff
@@ -0,0 +1,151 @@
+    public final PlannedDate plannedDate;
+    public final Recipe recipe;
+    private final String styleIngredientsAndSteps = "-fx-font-size: 11pt;\n"
+            + "-fx-font-family: \"Segoe UI\";\n"
+            + "-fx-text-fill: #FFFFFF;\n";
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label name;
+    @FXML
+    private Label date;
+    @FXML
+    private Label id;
+    @FXML
+    private ImageView favourite;
+    @FXML
+    private Label time;
+    @FXML
+    private VBox grains;
+    @FXML
+    private VBox vegetables;
+    @FXML
+    private VBox proteins;
+    @FXML
+    private VBox fruits;
+    @FXML
+    private VBox others;
+    @FXML
+    private VBox steps;
+    @FXML
+    private FlowPane goals;
+    @FXML
+    private Label ingredientsHeader;
+    @FXML
+    private Label stepsHeader;
+
+    public PlanningListCard(PlannedRecipe plannedRecipe, int displayedIndex) throws IOException {
+        super(FXML);
+        this.plannedRecipeObject = plannedRecipe;
+        this.plannedDate = plannedRecipe.getDate();
+        this.recipe = plannedRecipe.getRecipe();
+
+        date.setText(plannedDate.toString());
+
+        id.setText(displayedIndex + ". ");
+        name.setText(recipe.getName().fullName);
+        name.setWrapText(true);
+
+        if (recipe.isFavourite()) {
+            Path favouriteIconPath = Paths.get("src", "main", "resources", "images", "favourite.png");
+            favourite.setImage(new Image(Files.newInputStream(favouriteIconPath)));
+        }
+
+        recipe.getGoals().stream()
+                .sorted(Comparator.comparing(goal -> goal.goalName))
+                .forEach(goal -> goals.getChildren().add(new Label(goal.goalName)));
+
+        time.setText(recipe.getTime().value + " min");
+
+        ingredientsHeader.setText("Ingredients");
+        ingredientsHeader.setUnderline(true);
+        ingredientsHeader.setPadding(new Insets(10, 0, 0, 0));
+
+        recipe.getGrains().forEach(grain -> grains.getChildren().add(new Label(grain.toString())));
+        grains.getChildren().forEach(grain -> grain.setStyle(styleIngredientsAndSteps));
+
+        recipe.getVegetables().forEach(vegetable -> vegetables.getChildren().add(new Label(vegetable.toString())));
+        vegetables.getChildren().forEach(vegetable -> vegetable.setStyle(styleIngredientsAndSteps));
+
+        recipe.getProteins().forEach(protein -> proteins.getChildren().add(new Label(protein.toString())));
+        proteins.getChildren().forEach(protein -> protein.setStyle(styleIngredientsAndSteps));
+
+        recipe.getFruits().forEach(fruit -> fruits.getChildren().add(new Label(fruit.toString())));
+        fruits.getChildren().forEach(fruit -> fruit.setStyle(styleIngredientsAndSteps));
+
+        recipe.getOthers().forEach(other -> others.getChildren().add(new Label(other.toString())));
+        others.getChildren().forEach(other -> other.setStyle(styleIngredientsAndSteps));
+
+        stepsHeader.setText("Steps");
```

Comment:
could I suggest adding the `if (!recipe.getSteps().isEmpty()) { ... }`? Where `...` is line 118 to line 130. This is so that the header "Steps" will not appear if the recipe has no steps. You can refer to RecipeCard.java (under `seedu.recipe.ui`)!

Reply:
Alright! Thank you :) I will make the change.